figures/mapping-histology-labels.Rmd

```diff
@@ -0,0 +1,243 @@
+```
+
+# Take a look at how many biospecimens per `broad_histology` group
+
+Let's summarize `broad_histology`. 
+Because the `Normal` samples don't have histologies, we'll look at just the `Tumor` samples at for this summary. 
+
+```{r}
+broad_summary <- working_metadata %>% 
+  dplyr::filter(sample_type == "Tumor") %>%
+  dplyr::count(broad_histology) %>% 
+  dplyr::arrange(n)
+```
+
+Let's print out the summary. 
+
+```{r}
+broad_summary %>% 
+  knitr::kable()
+```
+
+There's handful of very small groups (many are n = 2). 
+
+## Declare new equivalent groups
+
+These groups we'll combine into a non-CNS/other tumor group.
+
+```{r}
+other_tumor <- c("lymphoma", "melanocytic tumor", "other tumor", "metastatic tumors", "non-cns tumor")
+```
+
+These groups we'll combine as a benign.
+
+```{r}
+benign <- c("benign tumor", "non-tumor")
+```
+
+Add in the `Other astrocytic tumor` in with the LGAT group. 
+
+```{r}
+lgat <- c("other astrocytic tumor", "low-grade astrocytic tumor")
+```
+
+# Make new `display_group`
+
+```{r}
+histology_table <- working_metadata %>% 
+  dplyr::mutate(
+    # NAs are really Normals
+    display_group = tidyr::replace_na(broad_histology, "normal"),
+    # Now do the group combining
+    display_group = forcats::fct_collapse(display_group,
+      "low-grade astrocytic tumor" = lgat,
+      "other tumor" = other_tumor,
+      "benign" = benign
+    ),
+    # Put this as a character for later handling
+    display_group = as.character(display_group)
+    )
+```
+
+Print out the number of `display_group` (including `normal`)!
+
+```{r}
+display_group_df <- histology_table %>% 
+  dplyr::count(display_group) %>% 
+  dplyr::arrange(n)
+
+knitr::kable(display_group_df)
+```
+
+Make this notebook stop if there are more than 16 histology groups + `Normal`. 
+
+```{r}
+if (nrow(display_group_df) > 17) {
+  stop("There's more than 16 histology `display_group` may want to re-evaluate the high-level histology groupings")
+}
+```
+
+# Add hex codes 
+
+These hex codes were retrieved from http://phrogz.net/css/distinct-colors.html with the settings on default for 18 colors.
+
+```{r}
+color_palette <- 
+  c("#ff0000", "#cc0000", "#995200", "#bfb300", "#fffbbf", 
+    "#2e7300", "#00e65c", "#00ffee", "#103d40", "#0085a6", 
+    "#003380", "#4073ff", "#737899", "#70008c", "#f2b6ee", 
+    "#ff40bf", "#8c0038", "#330d12"
+)
+```
+
+Declare how many colors we need. 
+
+```{r}
+n_colors <- nrow(display_group_df)
+```
+
+Make a named list color key where histologies are the names. 
+
+```{r}
+# Set seed so the colors are consistent upon re-run
+set.seed(2021)
+
+# Sample from the 18 colors
+subset_colors <- sample(color_palette, n_colors)
+names(subset_colors) <- display_group_df$display_group
+```
+
+We want `Other tumor` and the `Benign` group to both always be gray. 
+
+```{r}
+subset_colors[names(subset_colors) == 'other tumor'] <- "#D3D3D3"
+subset_colors[names(subset_colors) == 'benign'] <- "#808080"
```

Comment:
Looking at these, I would probably switch them on the assumption that "benign" is less important than "other", so we we might want "benign" to be lighter in color.

I also feel like "normal" should either not get a color or get a specific color? I don't know if we have any plots where we include the normal samples.

Reply:
I think they almost always get dropped (at least the plots I've made they do) but I'm open to suggestions if we have a color we'd like for the Normals.

Reply:
Normal biospecimens are never plotted, so I set the hex_code to NA.

Reply:
Nope. Switched to black.